docs/reference/index-modules.asciidoc

```diff
@@ -103,9 +103,11 @@ specific index module:
 
 `index.auto_expand_replicas`::
 
-    Auto-expand the number of replicas based on the number of available nodes.
+    Auto-expand the number of replicas based on the number of available data nodes.
+    Note that this does not take <<shard-allocation-filtering,shard allocation filtering rules>>
+    into account and can lead to a yellow cluster health.
     Set to a dash delimited lower and upper bound (e.g. `0-5`) or use `all`
-    for the upper bound (e.g. `0-all`).  Defaults to `false` (i.e. disabled).
+    for the upper bound (e.g. `0-all`). Defaults to `false` (i.e. disabled).
```

Comment:
How about

```
Automatically choose the number of replicas based on the number of data nodes in the cluster. Set to a dash delimited lower and upper bound (e.g. `0-5`) or use `all` for the upper bound (e.g. `0-all`). Defaults to `false` (i.e. disabled). Note that the number of replicas chosen does not take into account any other allocation rules such as <<shard-allocation-filtering,shard allocation filtering>> or `total_shards_per_node`, and this can lead to the cluster health becoming `YELLOW` if the applicable rules prevent all the replica shards from being allocated.
```